### PR TITLE
Remove deprecated std::iterator

### PR DESCRIPTION
**Pull request by ecatmur**
replace with direct typedefs

### CODE REVIEW

File: include/tins/sniffer.h [hunk 1]
```diff
@@ -449,8 +449,14 @@ HandlerProxy<T> make_sniffer_handler(T* ptr,
 /**
  * \brief Iterates over packets sniffed by a BaseSniffer.
  */
-class SnifferIterator : public std::iterator<std::forward_iterator_tag, Packet> {
+class SnifferIterator {
 public:
+    typedef std::forward_iterator_tag iterator_category;
```
Comment (neheb) [inline include/tins/sniffer.h:454]: Maybe replace typedef with using

Comment (ecatmur) [reply include/tins/sniffer.h]: OK, done

Comment (mfontanini) [reply include/tins/sniffer.h]: Unfortunately the default build mode still supports C++03 so we can't use `using` here and have to stick to typedef. It's probably about time to drop support for C++03 but we can't just yet.

**Comment by mfontanini** [comment]
Thanks and sorry for taking a year to reply!

**Pull request merged**